packages/xrpl/test/integration/submitAndWait.test.ts

```diff
@@ -1,91 +1,147 @@
+            responsePromise,
+            ledgerPromise,
+          ])
+
+          assert.equal(response.type, 'response')
+          assert.equal(response.result.validated, true)
+          retries = 0
+          break
+        } catch (err) {
+          const errorCodeRegex = /(?:Preliminary result:\s)(?<errorCode>.*)$/gu
+          const message = err.message as string
+          const matches = errorCodeRegex.exec(message)
+          const errorCode = matches?.groups?.errorCode
+
+          // Retry if another transaction finished before this one
+          // eslint-disable-next-line max-depth -- Testing
+          if (['tefPAST_SEQ', 'tefMAX_LEDGER'].includes(errorCode || '')) {
+            // eslint-disable-next-line no-await-in-loop, no-promise-executor-return -- We are waiting on retries
+            await new Promise((resolve) => setTimeout(resolve, 1000))
+          } else {
+            retries = 0
+            break
+          }
+        }
+      }
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'should throw a ValidationError when submitting an unsigned transaction without a wallet',
+    async () => {
+      const accountSet: AccountSet = {
+        TransactionType: 'AccountSet',
+        Account: testContext.wallet.classicAddress,
+        Domain: convertStringToHex('example.com'),
+      }
+
+      await assertRejects(
+        testContext.client.submitAndWait(accountSet),
+        ValidationError,
+        'Wallet must be provided when submitting an unsigned transaction',
+      )
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'submitAndWait a signed transaction',
+    async () => {
+      const accountSet: AccountSet = {
+        TransactionType: 'AccountSet',
+        Account: testContext.wallet.classicAddress,
+        Domain: convertStringToHex('example.com'),
+      }
+      const { tx_blob: signedAccountSet } = testContext.wallet.sign(
+        await testContext.client.autofill(accountSet),
+      )
+      const responsePromise = testContext.client.submitAndWait(signedAccountSet)
+      const ledgerPromise = delayedLedgerAccept(testContext)
+      return Promise.all([responsePromise, ledgerPromise]).then(
+        ([response, _ledger]) => {
+          assert.equal(response.type, 'response')
+          assert.equal(response.result.validated, true)
+        },
+      )
+    },
+    TIMEOUT,
+  )
+
+  it(
+    'submitAndWait a signed transaction longer',
+    async () => {
+      const accountSet: AccountSet = {
+        TransactionType: 'AccountSet',
+        Account: testContext.wallet.classicAddress,
+        Domain: convertStringToHex('example.com'),
+      }
+      const { tx_blob: signedAccountSet } = testContext.wallet.sign(
+        await testContext.client.autofill(accountSet),
+      )
+      const responsePromise = testContext.client.submitAndWait(signedAccountSet)
+      const ledgerPromise = delayedLedgerAccept(testContext)
+      return Promise.all([responsePromise, ledgerPromise]).then(
```

Comment:
Nice :) - I was struggling to find a good way to do the simultaneous `ledgerAccept` - this is very easy to read :)